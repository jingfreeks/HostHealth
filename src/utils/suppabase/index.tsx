export {default as supabase} from './suppabase';
