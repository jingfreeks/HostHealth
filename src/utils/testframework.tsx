import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
// import {QueryClient, QueryClientProvider} from '@tanstack/react-query';
// // import {DARK_THEME} from '@/utils/constants';
// // import {ThemeProvider} from 'styled-components/native';

type MockProviderType = {
  children?: React.ReactNode;
  store?: object;
};
export const MockProvider = (props: MockProviderType) => {
  const {store, children} = props;
  const mockStore = configureStore([thunk]);
  const mocked = mockStore(store);
  return <Provider store={mocked}>{children}</Provider>;
};

export const testingProps = (reference: string): unknown => ({
  testID: reference,
  accessibilityLabel: reference,
});

// const queryClient = new QueryClient({
//   defaultOptions: {
//     queries: {
//       retry: false,
//     },
//   },
// });
// export const wrapper = ({children}) => (
//   <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
// );
