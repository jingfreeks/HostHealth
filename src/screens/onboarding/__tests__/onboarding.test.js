import React from 'react';
import mockAsyncStorage from '@react-native-async-storage/async-storage/jest/async-storage-mock';
import {renderWithProviders} from '@/utils/testframeworknew';
import {waitFor} from '@testing-library/react-native';
import Profile from '../Profile';

jest.useFakeTimers();
jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      // eslint-disable-next-line @typescript-eslint/no-empty-function
      navigate: () => {},
    }),
    useIsFocused: () => true,
    useDispatch: () => ({dispatch: jest.fn()}),
  };
});
jest.mock('@supabase/supabase-js');
jest.mock('@react-native-async-storage/async-storage', () => mockAsyncStorage);



describe('On boarding profile Screen', () => {
  it('Should work as expected to get snapshot', () => {
    const all = renderWithProviders(<Profile />);
    waitFor(() => {
      expect(all.toJSON()).toMatchSnapshot();
    });
  });
});
