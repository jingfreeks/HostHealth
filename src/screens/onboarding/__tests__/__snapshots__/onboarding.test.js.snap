// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`On boarding profile Screen Should work as expected to get snapshot 1`] = `
<View
  style={
    [
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "marginBottom": 10,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "justifyContent": "center",
          "marginTop": 20,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "marginLeft": 20,
          },
        ]
      }
    >
      <View
        style={
          {
            "backgroundColor": "transparent",
            "borderRadius": 65,
            "height": 130,
            "width": 130,
          }
        }
        theme={
          {
            "colors": {
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": {
                "android": {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
          }
        }
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            {
              "backgroundColor": "transparent",
              "borderRadius": 65,
              "flex": 1,
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <Image
            borderRadius={65}
            onLoad={[Function]}
            source={
              {
                "uri": "https://s3.amazonaws.com/uifaces/faces/twitter/ladylexy/128.jpg",
              }
            }
            style={
              {
                "bottom": 0,
                "flex": 1,
                "height": undefined,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": undefined,
              }
            }
            testID="RNE__Image"
            theme={
              {
                "colors": {
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": {
                    "android": {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
              }
            }
            transition={true}
            transitionDuration={360}
          />
          <View
            accessibilityElementsHidden={true}
            collapsable={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            style={
              {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#bdbdbd",
                  "flex": 1,
                  "height": undefined,
                  "justifyContent": "center",
                  "width": undefined,
                }
              }
              testID="RNE__Image__placeholder"
            >
              <Text
                style={
                  {
                    "backgroundColor": "transparent",
                    "color": "#ffffff",
                    "fontSize": 65,
                    "textAlign": "center",
                  }
                }
              >
                LW
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "flex": 1,
                "height": undefined,
                "width": undefined,
              }
            }
            testID="RNE__Image__children__container"
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        },
      ]
    }
  >
    <RCTSafeAreaView
      style={
        [
          {},
        ]
      }
    >
      <View
        alignType="flex-start"
        style={
          [
            {
              "alignItems": "flex-start",
              "alignSelf": "flex-start",
              "flexDirection": "column",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#0e3f6c",
                "fontFamily": "Avenir",
                "fontSize": 18,
                "fontWeight": "900",
              },
            ]
          }
        >
          First Name
        </Text>
        <View
          style={
            [
              {
                "flexDirection": "row",
              },
            ]
          }
        >
          <TextInput
            autoCapitalize="none"
            onBlur={[Function]}
            onChangeText={[Function]}
            placeholder="First Name"
            secureTextEntry={false}
            style={
              [
                {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "height": 30,
                  "marginBottom": 5,
                },
              ]
            }
            value=""
          />
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#C2D6DC",
                  "fontSize": 25,
                },
                undefined,
                {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            î­Œ
          </Text>
        </View>
        <View
          style={
            [
              {
                "alignSelf": "stretch",
                "borderColor": "#C2D6DC",
                "borderWidth": 0.5,
                "flexDirection": "row",
                "marginBottom": 15,
              },
            ]
          }
        />
      </View>
    </RCTSafeAreaView>
  </View>
  <View
    style={
      [
        {
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        },
      ]
    }
  >
    <RCTSafeAreaView
      style={
        [
          {},
        ]
      }
    >
      <View
        alignType="flex-start"
        style={
          [
            {
              "alignItems": "flex-start",
              "alignSelf": "flex-start",
              "flexDirection": "column",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#0e3f6c",
                "fontFamily": "Avenir",
                "fontSize": 18,
                "fontWeight": "900",
              },
            ]
          }
        >
          Last Name
        </Text>
        <View
          style={
            [
              {
                "flexDirection": "row",
              },
            ]
          }
        >
          <TextInput
            autoCapitalize="none"
            onBlur={[Function]}
            onChangeText={[Function]}
            placeholder="Last Name"
            secureTextEntry={false}
            style={
              [
                {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "height": 30,
                  "marginBottom": 5,
                },
              ]
            }
            value=""
          />
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#C2D6DC",
                  "fontSize": 25,
                },
                undefined,
                {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            î­Œ
          </Text>
        </View>
        <View
          style={
            [
              {
                "alignSelf": "stretch",
                "borderColor": "#C2D6DC",
                "borderWidth": 0.5,
                "flexDirection": "row",
                "marginBottom": 15,
              },
            ]
          }
        />
      </View>
    </RCTSafeAreaView>
  </View>
  <View
    style={
      [
        {
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        },
      ]
    }
  >
    <RCTSafeAreaView
      style={
        [
          {},
        ]
      }
    >
      <View
        alignType="flex-start"
        style={
          [
            {
              "alignItems": "flex-start",
              "alignSelf": "flex-start",
              "flexDirection": "column",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#0e3f6c",
                "fontFamily": "Avenir",
                "fontSize": 18,
                "fontWeight": "900",
              },
            ]
          }
        >
          Middle Name
        </Text>
        <View
          style={
            [
              {
                "flexDirection": "row",
              },
            ]
          }
        >
          <TextInput
            autoCapitalize="none"
            onBlur={[Function]}
            onChangeText={[Function]}
            placeholder="Address"
            secureTextEntry={false}
            style={
              [
                {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "height": 30,
                  "marginBottom": 5,
                },
              ]
            }
            value=""
          />
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#C2D6DC",
                  "fontSize": 25,
                },
                undefined,
                {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            î­Œ
          </Text>
        </View>
        <View
          style={
            [
              {
                "alignSelf": "stretch",
                "borderColor": "#C2D6DC",
                "borderWidth": 0.5,
                "flexDirection": "row",
                "marginBottom": 15,
              },
            ]
          }
        />
      </View>
    </RCTSafeAreaView>
  </View>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "marginRight": 5,
          },
        ]
      }
    />
    <View
      border={10}
      borderc="#99E1E1"
      borderw={2}
      style={
        [
          {
            "borderColor": "#99E1E1",
            "borderRadius": 10,
            "borderWidth": 2,
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <View
        accessibilityLabel="AtomsButtonTestId"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#d6f3f3",
            "borderRadius": 10,
            "opacity": 1,
            "paddingBottom": 15,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 15,
          }
        }
        testID="AtomsButtonTestId"
      >
        <Text
          style={
            [
              {
                "color": "#03606A",
                "fontFamily": "Avenir",
                "fontSize": 15,
                "fontWeight": "500",
              },
            ]
          }
        >
          Next
        </Text>
      </View>
    </View>
  </View>
</View>
`;
